homeassistant/components/fastdotcom/coordinator.py

```diff
@@ -0,0 +1,31 @@
+"""DataUpdateCoordinator for the Fast.com integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from fastdotcom import fast_com
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DEFAULT_INTERVAL, DOMAIN, LOGGER
+
+
+class FastdotcomDataUpdateCoordindator(DataUpdateCoordinator):
+    """Class to manage fetching Fast.com data API."""
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize the coordintor for Fast.com."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(minutes=DEFAULT_INTERVAL),
+        )
+
+    async def _async_update_data(self) -> dict[str, str]:
+        """Run an executor job to retrieve Fast.com data."""
+        try:
+            return await self.hass.async_add_executor_job(fast_com)
+        except Exception as exc:
```

Comment:
Does the lib specify the exception?

Reply:
This is a rather old library; it doesn't throw a proper exception at all.